File: 2026-2/sesion-de-clase-3/lab01-2026-1/Question.java

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public abstract class Question implements Printable, Answerable {

    private int codigo;
    private String prompt;
    private List<String> opciones;

    public Question(int codigo, String prompt,
            List<String> opciones) {

        this.codigo = codigo;
        this.prompt = prompt;
        this.opciones = new ArrayList<>(opciones);
    }

    public int getCodigo() {
        return codigo;
    }

    public String getPrompt() {
        return prompt;
    }

    public List<String> getOpciones() {
        return Collections.unmodifiableList(opciones);
    }

    @Override
    public void devolverDatos() {
        System.out.println(prompt);
        int index = 0;
        for (String opcion : opciones) {
            System.out.println(
                    String.format("%d. %s", ++index, opcion));
        }
    }
}

File: 2026-2/sesion-de-clase-4/dotnet/Lab01-2026-1/Question.cs
namespace Lab01_2026_1;

public abstract class Question : Printable, Answerable
{
    private readonly List<string> opciones;

    public int Codigo { get; }
    public string Prompt { get; }
    public IList<string> Opciones => opciones.AsReadOnly();

    public Question(int codigo, string prompt, List<string> opciones)
    {
        Codigo = codigo;
        Prompt = prompt;
        this.opciones = [.. opciones];
    }

    public virtual void DevolverDatos()
    {
        Console.WriteLine(Prompt);
        int index = 0;
        foreach (string opcion in opciones)
        {
            index++;
            Console.WriteLine($"{index}. {opcion}");
        }
    }

    public abstract void RecuperarRespuesta(AssessmentItem item);
}
